docs/elastic_search.md

```diff
@@ -59,6 +59,172 @@ Consider using some combination of
+- Handles mapping conflicts by requiring explicit migration steps
+
+### es_reindex Command
+
+The `es_reindex` command populates Elasticsearch indices with data from the database. It supports incremental updates, performance optimization, and selective reindexing.
+
+#### Basic Usage
+
+```bash
+# Reindex all document types
+./manage.py es_reindex
+
+# Reindex specific document types
+./manage.py es_reindex --limit WikiDocument QuestionDocument
+
+# Reindex a percentage of documents (useful for testing)
+./manage.py es_reindex --percentage 10
+
+# Reindex a specific number of documents
+./manage.py es_reindex --count 1000
+```
+
+#### Time-Based Filtering
+
+```bash
+# Reindex documents updated after a specific date
+./manage.py es_reindex --updated-after "2024-01-01"
+
+# Reindex documents updated before a specific date
+./manage.py es_reindex --updated-before "2024-01-01 12:00:00"
+
+# Reindex documents updated within a date range
+./manage.py es_reindex --updated-after "2024-01-01" --updated-before "2024-01-02"
+```
+
+#### Performance Tuning
+
+```bash
+# Adjust chunk sizes for performance
+./manage.py es_reindex --sql-chunk-size 500 --elastic-chunk-size 100
+
+# Set custom timeout for bulk operations
+./manage.py es_reindex --timeout 60
+
+# Monitor SQL query performance
+./manage.py es_reindex --print-sql-count --count 100
+```
+
+#### Production Reindexing
+
+For production environments, use `CELERY_TASK_ALWAYS_EAGER=True` to process tasks synchronously:
+
+```bash
+# Reindex all document types in production
+CELERY_TASK_ALWAYS_EAGER=True ./manage.py es_reindex --limit WikiDocument
+CELERY_TASK_ALWAYS_EAGER=True ./manage.py es_reindex --limit QuestionDocument
+CELERY_TASK_ALWAYS_EAGER=True ./manage.py es_reindex --limit AnswerDocument
+CELERY_TASK_ALWAYS_EAGER=True ./manage.py es_reindex --limit ProfileDocument
+CELERY_TASK_ALWAYS_EAGER=True ./manage.py es_reindex --limit ForumDocument
+```
+
+#### How It Works
+
+- Queries the database for objects to index using Django ORM
+- Processes documents in configurable chunks to avoid memory issues
+- Sends bulk requests to Elasticsearch for efficient indexing
+- Uses Celery tasks for background processing (unless `CELERY_TASK_ALWAYS_EAGER=True`)
+- Supports both full reindexing and incremental updates based on modification dates
+
+### Common Workflows
+
+#### Creating New Indices
+
+When you need to create completely new indices (e.g., for mapping changes that require reindexing):
+
+```bash
+# 1. Create new index and switch writes
+./manage.py es_init --migrate-writes
+
+# 2. Populate the new index
+./manage.py es_reindex
+
+# 3. Switch reads to new index (zero downtime)
+./manage.py es_init --migrate-reads
+```
+
+#### Updating Existing Indices
+
+For compatible mapping changes that don't require full reindexing:
+
+```bash
+# Update mapping on existing index
+./manage.py es_init
+
+# Optionally reindex recently updated documents
+./manage.py es_reindex --updated-after "2024-01-01"
```

Comment:
I would use something like <<timeststamp before the reindex>> or something to indicate the slice